### PR TITLE
Zustand store

### PR DESCRIPTION
**Pull request by ella-blaupause**

### CODE REVIEW

File: stores/solvedPuzzleStore.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { create } from "zustand";
+import { initialSolvedPuzzles } from "../utils/utils";
+
+const useSolvedPuzlleStore = create((set) => ({
```
Comment (Satttoshi) [inline stores/solvedPuzzleStore.js:4]: useSolvedPuzzleStore*

File: stores/chatGPTStore.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { create } from "zustand";
```
Comment (Satttoshi) [inline stores/chatGPTStore.js:1]: `import {create as createStore} from "zustand";`

Can remove possible linting errors.

File: stores/chatGPTStore.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+const useChatGPTStore = create((set) => ({
+  questionInput: "",
+  isAnswered: false,
+
+  cacheQuestionInput: (event) => set({ questionInput: event.target.value }),
+  setIsAnsweredToTrue: () => set({ isAnswered: true }),
+  resetIsAnswered: () => set({ isAnswered: false }),
```
Comment (Satttoshi) [inline stores/chatGPTStore.js:9]: This 2 functions are redundant. You can do instead:
```js
setIsAnswered: (bool) => set({ isAnswered: bool }),
```

File: stores/colorCodePuzzleStore.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      colorCounts: { first: state.colorCounts.first++, ...state.colorCounts },
+    })),
+  increaseSecondColorDiv: () =>
+    set((state) => ({
+      colorCounts: { second: state.colorCounts.second++, ...state.colorCounts },
+    })),
+  increaseThirdColorDiv: () =>
+    set((state) => ({
+      colorCounts: { third: state.colorCounts.third++, ...state.colorCounts },
+    })),
```
Comment (Satttoshi) [inline stores/colorCodePuzzleStore.js:44]: These three functions are redundant and can be replaced with:

```js
increaseColorDiv: (color) =>
  set((state) => ({
    colorCounts: {
      ...state.colorCounts,
      [color]: state.colorCounts[color] + 1,
    },
  })),
```

You can call this function with:

```js
increaseColorDiv("first");
increaseColorDiv("second");
increaseColorDiv("third");
```

File: stores/solvedPuzzleStore.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+      solvedPuzzles: state.solvedPuzzles.map((solvedPuzzle) =>
+        solvedPuzzle.id === currentSolvedPuzzleId
+          ? { ...solvedPuzzle, isSolved: true }
+          : solvedPuzzle
+      ),
+    })),
+  resetSolvedPuzzle: () => set({ solvedPuzzles: initialSolvedPuzzles }),
+}));
+
+export default useSolvedPuzlleStore;
```
Comment (Satttoshi) [inline stores/solvedPuzzleStore.js:17]: useSolvedPuzzleStore*

File: components/MyButton/index.js [hunk 6]
```diff
@@ -1,4 +1,12 @@
+import { useRouter } from "next/router";
 import styled from "styled-components";
+import useLightStore from "../../stores/lightStore";
+import usePuzzlePiecesStore from "../../stores/puzzlePiecesStore";
+import useColorCodePuzzleStore from "../../stores/colorCodePuzzleStore";
+import useCrosswordStore from "../../stores/crosswordStore";
+import useToastStore from "../../stores/toastStore";
+import useChatGPTStore from "../../stores/chatGPTStore";
+import useSolvedPuzlleStore from "../../stores/solvedPuzzleStore";
```
Comment (Satttoshi) [inline components/MyButton/index.js:9]: In general we were recommended to only use 1 store at max. Anyways I like your structural approach and it makes alot of sense to me. I did not understand why Dominik said we have to only use a single Store, maybe you can ask that. For me i would keep it like that, its semantic and structural good job

**Review by Satttoshi — CHANGES_REQUESTED** [review]
I like your code alot, it is very semantic, clean written and good to understand. Keep that up!

Anyways I do have some suggestions you should address.

File: components/MyButton/index.js (hunk 6 above)
Comment (ella-blaupause) [reply components/MyButton/index.js]: thanks for the reminder, now I have implemented it👍

**Comment by Satttoshi** [comment]
LGTM :rocket:

**Pull request merged**